lua/astrocommunity/pack/rust/init.lua

```diff
@@ -25,32 +49,40 @@ return {
     end,
   },
   {
-    "nvim-treesitter/nvim-treesitter",
+    "jay-babu/mason-nvim-dap.nvim",
     opts = function(_, opts)
-      if opts.ensure_installed ~= "all" then
-        opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, "rust")
-      end
+      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, "codelldb")
     end,
   },
   {
-    "vxpm/ferris.nvim",
-    lazy = true,
-    init = function()
-      vim.api.nvim_create_autocmd("LspAttach", {
-        group = vim.api.nvim_create_augroup("lazy_ferris", { clear = true }),
-        desc = "Lazy load Ferris",
-        callback = function(args)
-          if vim.lsp.get_client_by_id(args.data.client_id).name == "rust_analyzer" then
-            if require("ferris.private.config").opts.create_commands then
-              require("ferris").create_commands(args.buf)
-            end
-          end
-        end,
-      })
+    "mrcjkb/rustaceanvim",
+    version = "^3",
+    ft = "rust",
+    opts = function()
+      local adapter
+      local success, package = pcall(function() return require("mason-registry").get_package "codelldb" end)
```

Comment:
We should propably make the dap configuration optional based upon whetever `nvim-dap` is enabled or not in the user's config, no?

Reply:
Yup! Updated in latest commit and also made other stuff `optional` as well, totally slipped my mind.

Reply:
Actually, I looked into the codebase and the dap stuff is already optional. we are setting up the adapter but rustaceanvim handles the optional nature of dap. So this is already good to go actually.